### ISSUE TITLE
Template for x-www-form-urlencoded

### ISSUE THREAD

**Issue by kymikoloco** [tcplugins/tcWebHooks#120]
## Expected Behavior
Some API endpoints don't take application/json payloads, and there needs to be a way to send different content types to them (x-www-form-urlencoded, namely)

## Current Behavior
Get a cryptic error from the API endpoint claiming the content wasn't sent.

## Steps to Reproduce (for bugs)

1. Sign up for chat.zulip.org
2. Create an *incoming webhook* bot, taking note of the API key and email
3. Create a new json template
4. Add a Build Event Template with Build Successful and the following JSON code
```json 
{
    "type": "private",
    "to": "youremail@example.com",
    "content":"${buildStatus}"
}
```
5. Create a new WebHook for a project with the url `https://chat.zulip.org/api/v1/messages`
6. Under **Extra Config** set to Basic Auth with the Username the email address of the bot and password the API Key of the bot
7. Under **Preview & Test** send a test webhook event
8. Bad Request (400) should be sent out. Debug Logs should say `{"var_name":"type","result":"error","code":"REQUEST_VARIABLE_MISSING","msg":"Missing 'type' argument"}`

## Your Environment

* tcWebHooks Version: 1.1-RC2.245.377 
* tcWebHooks REST API Version: 1.1-RC2.245.377 
* TeamCity Version: TeamCity Professional 2018.1.2 (build 58537)
* TeamCity server Operating System: Ubuntu 18.04.1 LTS x86_64
* Are you using a WebHook Template?: Yes, created from help in #115 

#### /opt/.BuildServer/config/webhook-templates.xml
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<webhook-templates>
    <webhook-template id="zulip-message" enabled="true" rank="21" format="nvpairs">
        <template-description>Send a Zulip message to a stream or a user</template-description>
        <template-tool-tip>POSTs a very simple zulip message</template-tool-tip>
        <preferred-date-format></preferred-date-format>
        <templates max-id="3">
            <template id="0">
      <!-- template does nothing for application/x-www-form-urlencoded right now -->
                <template-text use-for-branch-template="false">{
    "type": "private",
    "to": "email@example.com",
"content":"${buildStatus}"
}</template-text>
                <branch-template-text>{
    "type": "private",
    "to": "email@example.com",
"content":"${buildStatus}"
}</branch-template-text>
                <states>
                    <state type="buildStarted" enabled="true"/>
                    <state type="buildSuccessful" enabled="true"/>
                </states>
            </template>
        </templates>
    </webhook-template>
</webhook-templates>
```
#### opt/.BuildServer/config/projects/Nightly/pluginData/plugin-settings.xml
```xml
<?xml version="1.0" encoding="UTF-8"?>
<settings>
  <webhooks enabled="true">
    <webhook url="https://chat.zulip.org/api/v1/messages" enabled="true" format="nvpairs" template="zulip-message">
      <states>
        <state type="buildStarted" enabled="true" />
        <state type="beforeBuildFinish" enabled="false" />
        <state type="responsibilityChanged" enabled="false" />
        <state type="changesLoaded" enabled="false" />
        <state type="buildInterrupted" enabled="false" />
        <state type="buildSuccessful" enabled="true" />
        <state type="buildBroken" enabled="false" />
        <state type="buildFinished" enabled="true" />
        <state type="buildFailed" enabled="false" />
        <state type="buildFixed" enabled="false" />
      </states>
      <build-types enabled-for-all="true" enabled-for-subprojects="false">
        <build-type id="bt8" />
      </build-types>
      <!-- Parameters here do nothing -->
      <parameters>
        <param name="content" value="${buildStatus}" />
        <param name="to" value="email@example.com" />
        <param name="type" value="private" />
      </parameters>
      <auth enabled="true" type="userpass" preemptive="true">
        <auth-parameters>
          <param name="username" value="TeamCity-bot@example.com" />
          <param name="password" value="API_KEY" />
          <param name="realm" value="" />
        </auth-parameters>
      </auth>
      <headers>
        <header name="Authorization" value="Basic base64stuff==" />
        <header name="Content-Type" value="application/x-www-form-urlencoded" />
      </headers>
    </webhook>
  </webhooks>
</settings>
```
#### /opt/.BuildServer/config/projects/Nightly/project-config.xml
Contains the parameters for the Nightly Project that zulip expects and nvpairs will send
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" parent-id="Example" uuid="ab9a8dc4-8d23-48ea-84a5-c3c594ab2a54" xsi:noNamespaceSchemaLocation="http://www.jetbrains.com/teamcity/schemas/2018.1/project-config.xsd">
  <name>Nightly</name>
  <description>Nightly builder. Cleans all sources and deploys to artifact storage</description>
  <parameters>
    <param name="webhook.content" value="TeamCity version ${system.teamcity.version} running as user ${env.LOGNAME} - ${buildStatus}" />
    <param name="webhook.to" value="email@example.com" />
    <param name="webhook.type" value="private" />
  </parameters>
  <cleanup />
</project>
```

**Comment by netwolfuk** [comment]
This will be released as part of tcWebHooks version 1.2.x

**Comment by netwolfuk** [comment]
1.2.0 alpha.1 released which includes this fix.
